.spectral.yml

```diff
@@ -0,0 +1,73 @@
+extends: spectral:oas
+rules:
+  oas3-unused-components-schema: off
+  tag-description: off
+  oas3-api-servers: off
+  openapi-tags: off
+  info-description: off
+  schema-names-pascal-case:
+    description: Schema names MUST be written in PascalCase
+    message: '{{property}} is niet PascalCase: {{error}}'
+    recommended: true
+    type: style
+    given: '$.components.schemas.*~'
+    then:
+      function: casing
+      functionOptions:
+        type: pascal
```

Comment:
Gevolg is dat alle *_links, *_embedded, *_enum en *_tabel componenten een waarschuwing geven. Het opnemen van een underscore is namelijk niet in lijn met UpperCamelCase/PascalCase.

Vanuit consistentie zou ik het ook beter vinden deze underscore weg te halen.

Kan dat nu nog zonder grote consequenties? In BAG en BRK hebben we ook underscore in de schemacomponent namen gebruikt (dan ontstaat er weer inconsistentie tussen de Haal Centraal API's). Of kunnen we die ook aanpassen in de eerstvolgende release? M.a.w. levert dat breaking changes op bij iemand?

Reply:
Bij *_links, *_embedded, *_tabel leidt het tot een breaking change bij de providers, niet bij de consumers omdat de consumers niets doen met de classes zelf. Voorwaarde is alleen hernoemen van de schema naam en niet property namen.
Voor de _enums zal ik testen

Reply:
_links en _embedded zijn reserved words in hal+json. Als je de underscores weghaalt dan wordt het plain json en moet je het content-type erop aanpassen. Dit is een afweging. Vind je consistentie in de naamgeving van properties belangrijker dan het voldoen aan de draft van json hal. Mijn voorkeur gaat naar het handhaven van de underscores bij _links en _embedded.

Reply:
> 
> 
> _links en _embedded zijn reserved words in hal+json. Als je de underscores weghaalt dan wordt het plain json en moet je het content-type erop aanpassen. Dit is een afweging. Vind je consistentie in de naamgeving van properties belangrijker dan het voldoen aan de draft van json hal. Mijn voorkeur gaat naar het handhaven van de underscores bij _links en _embedded.

Even een check. @fsamwel, je hebt hier alleen over de component namen toch? Niet over de properties, want hernoemen van propertynamen leidt tot breaking changes bij zowel consumers als providers.
Ik ben het dus eens met @JohanBoer om _embedded en _links property namen niet te veranderen en ook hal json te blijven als content type.
Ik ga kijken of de jsonpath kan worden aangepast zodat de _links en _embedded properties niet worden meegenomen in de casing check

Reply:
ja, het gaat hier om de componentnamen, niet om de propertynamen.

Ik denk dat het in de validatie kan worden aangepast door toch niet de built in functie casino (Pascal) te gebruiken, maar zelf een reguliere expressie te maken die deze extensies wel toelaat:
/^([A-Z][a-z0-9]+)+(\_embedded|\_links|\_enum|\_tabel)?$/